bundles/org.openhab.binding.serial/src/main/java/org/openhab/binding/serial/internal/handler/SerialBridgeHandler.java

```diff
@@ -0,0 +1,342 @@
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.serial.internal.handler;
+
+import static org.openhab.binding.serial.internal.SerialBindingConstants.BINARY_CHANNEL;
+import static org.openhab.binding.serial.internal.SerialBindingConstants.STRING_CHANNEL;
+import static org.openhab.binding.serial.internal.SerialBindingConstants.TRIGGER_CHANNEL;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.charset.Charset;
+import java.nio.charset.IllegalCharsetNameException;
+import java.util.Base64;
+import java.util.TooManyListenersException;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.serial.internal.util.Parity;
+import org.openhab.binding.serial.internal.util.StopBits;
+import org.openhab.core.io.transport.serial.PortInUseException;
+import org.openhab.core.io.transport.serial.SerialPort;
+import org.openhab.core.io.transport.serial.SerialPortEvent;
+import org.openhab.core.io.transport.serial.SerialPortEventListener;
+import org.openhab.core.io.transport.serial.SerialPortIdentifier;
+import org.openhab.core.io.transport.serial.SerialPortManager;
+import org.openhab.core.io.transport.serial.UnsupportedCommOperationException;
+import org.openhab.core.library.types.RawType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.CommonTriggerEvents;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SerialBridgeHandler} is responsible for handling commands, which
+ * are sent to one of the channels.
+ *
+ * @author Mike Major - Initial contribution
+ */
+@NonNullByDefault
+public class SerialBridgeHandler extends BaseBridgeHandler implements SerialPortEventListener {
+
+    private final Logger logger = LoggerFactory.getLogger(SerialBridgeHandler.class);
+
+    private SerialBridgeConfiguration config = new SerialBridgeConfiguration();
+
+    private final SerialPortManager serialPortManager;
+    private @Nullable SerialPort serialPort;
+
+    private @Nullable InputStream inputStream;
+    private @Nullable OutputStream outputStream;
+
+    private Charset charset = Charset.defaultCharset();
```

Comment:
To have the same behavior across all platforms and configurations, I'd use a static charset here.
```suggestion
    private Charset charset = StandardCharsets.UTF_8;
```